### PR TITLE
🧚🏿‍♂️ Form modal

### PR DESCRIPTION
**Pull request by vasuchal**
## Description
This is a modal that shows up when users want to fill out the form. 

Fixes #<number>

## Screenshots
Screen Shot 2021-04-08 at 9 42 27 PM

### CODE REVIEW

File: src/components/mobile/SettingsPage.js [hunk 1]
```diff
@@ -16,6 +17,10 @@ import { Slider } from '../ui/Slider'
 import { Tag, TagList } from '../ui/Tag'
 
 const SettingsPage = () => {
+  const [showDialog, setShowDialog] = useState(false)
+  const open = () => setShowDialog(true)
+  const close = () => setShowDialog(false)
+  const buttonRef = useRef()
```
Comment (Qrtn) [inline src/components/mobile/SettingsPage.js:23]: Why is this ref necessary?

File: src/components/mobile/SettingsPage.js [hunk 2]
```diff
@@ -16,6 +17,10 @@ import { Slider } from '../ui/Slider'
 import { Tag, TagList } from '../ui/Tag'
 
 const SettingsPage = () => {
+  const [showDialog, setShowDialog] = useState(false)
+  const open = () => setShowDialog(true)
+  const close = () => setShowDialog(false)
```
Comment (Qrtn) [inline src/components/mobile/SettingsPage.js:22]: You can pass these inline.

File: src/components/ui/Modal.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import '@reach/dialog/styles.css'
+
+import { Dialog } from '@reach/dialog'
+import styled from 'styled-components/macro'
+
+const StyledDialog = styled(Dialog)`
+  border-radius: 23px;
+  width: 75%;
+  height: 70%;
```
Comment (Qrtn) [inline src/components/ui/Modal.js:9]: Is the padding correct on the dialog?

**Review by Qrtn — CHANGES_REQUESTED** [review]
Nice work Vasu!

File: src/components/mobile/SettingsPage.js (hunk 1 above)
Comment (vasuchal) [reply src/components/mobile/SettingsPage.js:23]: I implemented the example from Reach UI to get the buttons to open and close the modal and they had refs but it seems to work without so I'll remove em!

File: src/components/ui/Modal.js (hunk 3 above)
Comment (vasuchal) [reply src/components/ui/Modal.js:9]: I looked on the figma and I couldn't find a padding property so I am not too sure how to handle this

Comment (Qrtn) [reply src/components/ui/Modal.js:9]: Messaged you about this! But will include here as well for other ppl:

- A lot of layout CSS generated by Figma isn't too reliable, except for font sizes, shadows, colors, etc. Figma tends to generate absolute top, left, bottom, right measurements, which doesn't reflect that you should use padding
- Your goal is always to try to make your component **visually** similar to the design. Figma's CSS is just a tool to help you achieve that. Right now, it looks like there's too much space between the border and the text. How can you fix that?

File: src/components/ui/Modal.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import '@reach/dialog/styles.css'
+
+import { Dialog } from '@reach/dialog'
+import styled from 'styled-components/macro'
+
+const Modal = styled(Dialog)`
+  border-radius: 23px;
+  padding: 23px 17px 21px 17px;
```
Comment (Qrtn) [inline src/components/ui/Modal.js:8]: The padding should probably be equal on the top and bottom.

File: src/components/ui/Modal.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import styled from 'styled-components/macro'
 
 const Modal = styled(Dialog)`
   border-radius: 23px;
-  padding: 23px 17px 21px 17px;
+  padding: 23px 17px 23px 17px;
```
Comment (Qrtn) [inline src/components/ui/Modal.js:8]: You can write this as `23px 17px`!

File: src/components/mobile/SettingsPage.js [hunk 6]
```diff
@@ -16,6 +17,7 @@ import { Slider } from '../ui/Slider'
 import { Tag, TagList } from '../ui/Tag'
 
 const SettingsPage = () => {
+  const [showDialog, setShowDialog] = useState(false)
   const [currentStep, setCurrentStep] = useState(2)
```
Comment (Archna-1) [inline src/components/mobile/SettingsPage.js:21]: What does the 2 mean  😮

**Review by Archna-1 — APPROVED** [review]
LGTM! :)

**Review by Qrtn — APPROVED** [review]
LGTM!

File: src/components/mobile/SettingsPage.js (hunk 6 above)
Comment (vasuchal) [reply src/components/mobile/SettingsPage.js:21]: it's the current step!

**Pull request merged**